CMakeLists.txt

```diff
@@ -23,65 +23,69 @@ add_subdirectory(src)
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/scripts/cmake)
 
 option(ADA_BENCHMARKS "Build benchmarks" OFF)
-
-include(cmake/CPM.cmake)
-# CPM requires git as an implicit dependency
-find_package(Git QUIET)
-# We use googletest in the tests
-if(Git_FOUND AND BUILD_TESTING)
-  CPMAddPackage(
-    NAME GTest
-    GITHUB_REPOSITORY google/googletest
-    VERSION 1.14.0
-    OPTIONS  "BUILD_GMOCK OFF" "INSTALL_GTEST OFF"
-  )
-endif()
-# We use simdjson in both the benchmarks and tests
-if(Git_FOUND AND (BUILD_TESTING OR ADA_BENCHMARKS))
-  CPMAddPackage("gh:simdjson/simdjson@3.3.0")
-endif()
-# We use Google Benchmark, but it does not build under several 32-bit systems.
-if(Git_FOUND AND ADA_BENCHMARKS AND (CMAKE_SIZEOF_VOID_P EQUAL 8))
-  CPMAddPackage(
-    NAME benchmark
-    GITHUB_REPOSITORY google/benchmark
-    GIT_TAG f91b6b4
-    OPTIONS "BENCHMARK_ENABLE_TESTING OFF"
-            "BENCHMARK_ENABLE_INSTALL OFF"
-            "BENCHMARK_ENABLE_WERROR OFF"
-
-  )
-endif()
-
-if (BUILD_TESTING AND NOT EMSCRIPTEN)
-  if(Git_FOUND)
-    message(STATUS "The tests are enabled.")
-    add_subdirectory(tests)
-  else()
-    message(STATUS "The tests are disabled because git was not found.")
+option(ADA_TESTING "Build tests" ${BUILD_TESTING})
+
+if(ADA_TESTING OR ADA_BENCHMARKS OR ADA_TOOLS)
```

Comment:
Can you add a comment to here to why this change occurred? It's important that we don't break embedders like you in the future.

Reply:
Done! Let me know if there is any other context I can add to be useful :)